addons/web/static/src/views/view_dialogs/select_create_dialog.js

```diff
@@ -0,0 +1,66 @@
+/** @odoo-module **/
+
+import { Dialog } from "@web/core/dialog/dialog";
+import { View } from "../view";
+import { useService } from "@web/core/utils/hooks";
+
+const { onWillStart, useState, xml } = owl;
+
+export class SelectCreateDialog extends Dialog {
+    setup() {
+        super.setup();
+        this.viewService = useService("view");
+        this.state = useState({
+            selectedRecords: [],
+        });
+
+        this.propsView = {
+            resModel: this.props.resModel,
+            domain: this.props.domain,
+            context: this.props.context,
+            type: "list",
+            editable: false,
+            showButtons: false,
+            hasSelectors: !this.props.disableMultipleSelection,
+            selectRecord: (resId) => {
```

Comment:
not sure about this, but maybe this should be the default mechanism, even for actionService.switchView (could be set by the action service itself to perform a switchView. That way, views no longer need to do a switchView themselves. To discuss...